### PR TITLE
refactor(editor): Remove editor theme providers etc

### LINKED ISSUE #2389: Remove theme config property from editor codebase
As the editor code is merged with frontend and there are no plans to make it highly customizable for other users, we can now simplify the styling by removing the `theme` config property and its usage from the editor codebase.

### PR DESCRIPTION
**Pull request by elbotho**
this PR removes all global theme code (providers, types, defaults…).

I replaced the values with some hardcoded variables in `colors.ts` (`legacyEditorTheme`) until the UI rework is done.

Status:
~`theme` is still used in some editor components. I will adapt some here, but it might makes sense to keep the theme config here for now until we actually rework those components.~

replaced everywhere.
let's see how many styles I broke 😇

### CODE REVIEW

**Comment by Entkenntnis** [comment]
grafik

font color looks off

File: src/serlo-editor-repo/plugin-text/components/inline-overlay.tsx [hunk 1]
```diff
@@ -44,13 +41,12 @@ const Triangle = styled.div<{ position: InlineOverlayPosition }>(
       borderLeft: '5px solid transparent',
       borderRight: '5px solid transparent',
       // eslint-disable-next-line @typescript-eslint/restrict-template-expressions, @typescript-eslint/no-unsafe-member-access
```
Comment (Entkenntnis) [inline src/serlo-editor-repo/plugin-text/components/inline-overlay.tsx]: Can we remove these comments?

File: src/serlo-editor-repo/plugin-text/components/text-editor.tsx [hunk 2]
```diff
@@ -351,7 +351,7 @@ export function TextEditor(props: TextEditorProps) {
           onKeyDown={handleEditableKeyDown}
           onPaste={handleEditablePaste}
           renderElement={renderElementWithEditorContext(config, focused)}
-          renderLeaf={renderLeafWithConfig(config)}
+          renderLeaf={renderLeaf()}
```
Comment (hejtful) [inline src/serlo-editor-repo/plugin-text/components/text-editor.tsx]:
```suggestion
          renderLeaf={renderLeaf}
```

File: src/serlo-editor-repo/plugin-text/components/text-editor.tsx [hunk 3]
```diff
@@ -413,9 +413,9 @@ function renderElementWithEditorContext(
   }
 }
 
-function renderLeafWithConfig(config: TextEditorConfig) {
+function renderLeaf() {
   return function renderLeaf(props: RenderLeafProps) {
```
Comment (hejtful) [inline src/serlo-editor-repo/plugin-text/components/text-editor.tsx]:
```suggestion
function renderLeaf(props: RenderLeafProps) {
```

**Review by Entkenntnis — APPROVED** [review]
played around a bit, one small note and one place where font color is wrong, otherwise looks good!

File: src/serlo-editor-repo/plugin-text/components/text-editor.tsx (hunk 2 above)
Comment (elbotho) [reply src/serlo-editor-repo/plugin-text/components/text-editor.tsx]: thanks!

File: src/serlo-editor-repo/plugin-text/components/inline-overlay.tsx (hunk 1 above)
Comment (elbotho) [reply src/serlo-editor-repo/plugin-text/components/inline-overlay.tsx]: nice, missed this one.

**Comment by elbotho** [comment]
thanks for the reviews!

**Review by hejtful — APPROVED** [review]

**Comment by hejtful** [comment]
Thank you for the refactor!

**Pull request merged**